samples/lib/thrift/hello_client/Makefile

```diff
@@ -15,7 +19,17 @@ GENOBJ = $(filter-out %.h, $(GENSRC:.cpp=.o))
 THRIFT_FLAGS :=
 THRIFT_FLAGS += $(shell pkg-config --cflags thrift)
 THRIFT_FLAGS += -I$(GEN_DIR)
-THRIFT_LIBS = $(shell pkg-config --libs thrift)
+THRIFT_LIBS :=
+THRIFT_LIBS += $(shell pkg-config --libs thrift)
+THRIFT_LIBS += $(shell pkg-config --libs zlib)
+THRIFT_LIBS += $(shell pkg-config --libs openssl)
+
+LIB_DIR = ../../../../.upstream/lib/cpp/src/thrift/
+LIBSRC :=
+LIBSRC += TSSLSocket.cpp
+LIBSRC += TZlibTransport.cpp
+
+LIBOBJ = $(patsubst %.cpp,$(GEN_DIR)/%.o,$(LIBSRC))
 
 all: hello_client
```

Comment:
It would likely make sense to have several Makefile targets that are built with 'all':
 * hello_client
 * hello_client_compact
 * hello_client_zlib
 * hello_client_ssl